src/v/cluster_link/manager.h

```diff
@@ -90,6 +98,14 @@ class manager {
 
     model::cluster_link_task_status_report get_task_status_report() const;
 
+    kafka::data::rpc::topic_metadata_cache*
+    topic_metadata_cache() const noexcept;
+
+    kafka::data::rpc::partition_leader_cache*
+    partition_leader_cache() const noexcept;
```

Comment:
Those methods should return reference instead of a naked pointer. We use pointers not to deal with references being class fields. Also the method are const but return a pointer which doesn't maintain constness